### DISCUSSION TITLE
2023-07-07 - Data Engineering #4 - Data Lakes - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by lukecookssw** [SSWConsulting/SSW.Rules.Content#5946] [Chewing the Fat]
Hey data engineers!

Let's talk about [this rule](https://ssw.com.au/rules/do-you-use-a-data-lake/)

The rule has this **Scenario**: Imagine you work at Northwind. They have all these systems:

- GitHub/Azure DevOps – issues, commits, etc.
- Billing - Timesheets, Client invoices, etc
- CRM – Projects, skills, technologies, sales, leads, etc.
- Accounting – Business expenses, client payment receipts, payment dates, etc.
- Office automation logs – employee arrival/leave times, device usage, security alerts, etc.
- Websites – traffic, analytics, etc.

Comment below for what you think would be:
1. A cool question would you ask this data? Be creative!
2. How many systems would you be interacting with?
3. Would you need a Data Lake or Data Warehouse?

**Comment by GordonBeeming** [comment]
`1. A cool question would you ask this data? Be creative!`
It would be cool to have suggested rules... the way I'd see this working is that by looking proportionally to where our billing and skills on those projects are, we should have rules that highlight those same areas. Looking at the recency of rules would taken into account too, so if we've got 400 rules on .net but they are all outdated, we should fix that. 

`2. How many systems would you be interacting with?`
4

`3. Would you need a Data Lake or Data Warehouse?`
It could start off using a data warehouse, although expanding on it might be easier down the line with a data lake

**Comment by drwharris** [comment]
EDIT: Updated to account for the change in question.

> A cool question would you ask this data? Be creative!

How used is a feature in comparison with the effort expended in development

> How many systems would you be interacting with?

 Source control and the end system

> Would you need a Data Lake or Data Warehouse?

Depends if any Transformation is needed. None, then a DW. else DL

OLD:
Maybe tie the github/azure stats into the "leaderboard" in SSW Times. Whos winning the PR or commit race? 
Also some info on what rules are getting viewed, by who?

**Reply by brydeno** [reply]
More interestingly, 

- do employees who work longer produce more?
- Do they produce code with less issues?
- Does this validate the external research around shorter working hours increasing productivity (ie the push towards a 4 day week)?

**Comment by tkapa** [comment]
> cool question would you ask this data? Be creative!
Productivity metrics of teams with projects on different platforms (Gh, AzDo)

> How many systems would you be interacting with?
4?

> Would you need a Data Lake or Data Warehouse?
Depending on scale, I'd likely start with a Warehouse and scale up.

**Comment by matt-goldman** [comment]
> A cool question would you ask this data? Be creative!

I would want to know whether we can draw any conclusions about how the intersection of all the factors that feed into this data impact productivity. It would be cool to see if we can identify for example environmental factors in the offices that affect this. We of course need to be aware that correlation does not equal causation - one obvious example is that it would be easy to look at the data and say that the more time we spend on internal products, the less money we earn (when in fact the causal relationship is the other way around). I'm more interested in what makes developers happy and comfortable. And obviously this impacts revenue and the profitability of the business. It would also be cool to see if we can identify client personas or verticals that yield the best outcomes, and this would help us focus our efforts on driving engagement in those areas, or shoring up the gaps.


> How many systems would you be interacting with?

All of them.



>Would you need a Data Lake or Data Warehouse?

Both. A Data Lake to aggregate all the data and feed into the analysis, and a Data Warehouse for reporting.

**Comment by ncn-ssw** [comment]
A cool question that could be asked to this data might be: "Can we predict a project's success based on historical timesheets, project data, associated skills and technologies, and correlating factors such as employee arrival/leave times, device usage, and website traffic during the project's timeframe?"

For this question, I'd be interacting with the following systems:

1. GitHub/Azure DevOps - to gather data on the issues and commits related to the project.
2. Billing - to collect timesheet data which may indicate project's progress and employee dedication.
3. CRM - for project details, associated skills and technologies.
4. Office automation logs - to analyze employee's work pattern and device usage during the project.
5. Websites - to get an understanding of the traffic and analytics related to the project or the product.

Considering the variety, volume, and the potential semi-structured nature of this data, we would likely benefit from using a Data Lake to store the raw data. Then, we could leverage a Data Warehouse to store the processed, cleaned, and structured data ready for analysis. 🤖⭐

**Comment by ozdemir-mehmet** [comment]
A cool question would you ask this data? Be creative!
How much does each PR cost, the group 
How many systems would you be interacting with?
Maybe 3
Would you need a Data Lake or Data Warehouse?
DataWarehouse would always be my end goal

**Comment by jakebayliss** [comment]
> A cool question would you ask this data? Be creative!

Show me SSW's commits per day over time 

> How many systems would you be interacting with?

2

> Would you need a Data Lake or Data Warehouse?

Warehouse

**Comment by calumjs** [comment]
Question | Systems Involved | Data Lake or Data Warehouse
-- | -- | --
Can we predict the likelihood of future sales leads based on the type of projects we are currently working on and the pattern of website traffic? | CRM 📊, Websites 🌐, GitHub/Azure DevOps 💻 | Data Warehouse 🏦 <br> (Mostly structured data involved)
How does the interaction of the technologies used in projects with the skills of the developers affect the billable hours and the resulting project profitability? | CRM 📊, GitHub/Azure DevOps 💻, Billing 💵, Accounting 📔 | Data Lake 🏞️ <br> (Unstructured data like commits or issues from DevOps needs to be analyzed)
Based on the patterns in website traffic and the types of technologies/skills involved in our projects, can we predict which technologies or skills will be in high demand in the future? | CRM 📊, Websites 🌐, GitHub/Azure DevOps 💻 | Data Warehouse 🏦 <br> (Mostly structured data involved for pattern analysis and trend prediction)

**Comment by Jord-Gui** [comment]
1. A cool question would you ask this data? Be creative!
How much money does it cost to maintain devs that don't do any work 😈

2. How many systems would you be interacting with?
4

3. Would you need a Data Lake or Data Warehouse?
Data Warehouse

**Comment by NadeekaJK** [comment]
Interesting stuff!!

**Comment by vladislav-kir** [comment]
1. A cool question would you ask this data? Be creative!
Do people who spend more time in the office are more or less productive? 

2. How many systems would you be interacting with?
2 - Use Office automation logs – employee arrival/leave times and GitHub/Azure DevOps – issues, commits...

3. Would you need a Data Lake or Data Warehouse?
Data Warehouse

**Comment by cvionnet** [comment]
Great video, it gives a better understanding of the role of Data Engineering and Data Scientist.

1. A cool question would you ask this data? Be creative!
"What is the correlation between employee arrival/leave times and their productivity based on GitHub/Azure DevOps commit patterns?"🤖

2. How many systems would you be interacting with?
Probably 3

3. Would you need a Data Lake or Data Warehouse?
It depends©
- If the company just wants to show off their existing data with fancy reports, a data warehouse should do the trick
- But if the company has a bunch of data scientists who like to play around and make predictions with data, throwing in a Data Lake would be a smart move

**Comment by tombui99** [comment]
> A cool question would you ask this data? Be creative!

Based on the CRM invoice and GitHub commits data, can we tell if there's a correlation of the productivity of a developer between when working internally and on client?

> How many systems would you be interacting with?

3

> Would you need a Data Lake or Data Warehouse?

It would be useful to have them if we plan on making fancy dashboards or make predictions

**Comment by 2fernandez** [comment]
![P1](https://github.com/SSWConsulting/SSW.Rules.Content/assets/27670973/cae9d366-abca-44c4-9418-01e21f44720a)

**Comment by AlexRothwell** [comment]
> A cool question would you ask this data? Be creative!

Is there a relationship between PR size and delivered value?

> How many systems would you be interacting with?

4

> Would you need a Data Lake or Data Warehouse?

Data lake - how value is defined will change based on project, context and stakeholder - being able to explore in different ways would be useful

**Comment by Harry-Ross** [comment]
> A cool question would you ask this data? Be creative!

I think it would be interesting to analyse how much time is spent on each PBI compared to the expected size of the PBI by members of the team. 

> How many systems would you be interacting with?

2 - (GitHub/Azure DevOps, Timesheets)

> Would you need a Data Lake or Data Warehouse?

I think it would be much more valuable to have processed data from a Data Warehouse.

**Comment by Khaled-Albahsh** [comment]
> A cool question would you ask this data? Be creative!
how do you sleep at night with all of this over your head?
     > How many systems would you be interacting with?
3
     > Would you need a Data Lake or Data Warehouse?
Data Lake - Manipulation of data is easier :)

**Comment by MarcusIrmscherSSW** [comment]
1. A cool question would you ask this data? Be creative!
What is the correlation between employee arrival/leave times and website traffic? 

2. How many systems would you be interacting with?
6

3. Would you need a Data Lake or Data Warehouse?
Data Lake and a Data Warehouse

**Comment by SamBWagner** [comment]
These CTFs are great for sparking interest and "high-level" understanding on the topics at hand. However, they don't do a great job of sparking lower level understanding. 

I.e. This CTF didn't mention the real difference between Databases, Data Warehouses and Data Lakes.  This is the type of learning I prefer/require for me to understand things so these CTFs take me quite a while to get through as I have to go and read quite a lot.

An excellent start though and I'm keen to either keep going with these or begin the process of learning on my own.

**Comment by apforsyth** [comment]
I learned a new term today, made me laugh - Data Swamp! ;0

**Comment by danielmackay** [comment]
> 1. A cool question would you ask this data? Be creative!

For a given project, taking into account all developers working on a single PBI, and their commits into source code, and time spent on the project how much has does each PBI cost?

> 2. How many systems would you be interacting with?

GItHub, CRM, Billing, and possibly Accounting

> 3. Would you need a Data Lake or Data Warehouse?

As this is a specific question that we know ahead of time, I would use a Data Warehouse to gather the data we need for this question and structure it appropriately.

**Comment by wicksipedia** [comment]
> A cool question would you ask this data? Be creative!

Can we establish a relationship between the frequency of commits on GitHub and the subsequent billable hours, factoring in the complexity of projects and employee skills, to optimize project timelines and cost?

![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/600044/6712ff4e-d641-47fe-b959-bd7f02f1df7c)


> How many systems would you be interacting with?

5
- GitHub
- Billing
- CRM
- Accounting
- Office automation logs

> Would you need a Data Lake or Data Warehouse?

Both - a hybrid approach.

A data lake could be used for storing all raw data and exploratory data analysis, and a data warehouse could be used for structured, regular reports and analyses. This allows us to leverage the advantages of both while mitigating their individual limitations.

**Comment by AntPolkanov** [comment]
> 1. A cool question would you ask this data? Be creative!

Find insights and correlations between website activity and sales. 

> 2. How many systems would you be interacting with?

Website, CRM, Accounting.

> 3. Would you need a Data Lake or Data Warehouse?

Date Lake. A data pipeline would extract the data and load them. Then it is consumed by data scientists.

**Comment by bradystroud** [comment]
We could add some features to https://timeband.com.au/

**Comment by TobyChurches** [comment]
`>1. A cool question would you ask this data? Be creative!`
The relationship between Expected and Actual Effort
`>2. How many systems would you be interacting with?`
2?
`>3. Would you need a Data Lake or Data Warehouse?`
Probably a Data Warehouse

**Comment by jeoffreyfischer** [comment]
I found the video enlightening, it describes the complexity and abstraction of data lake processing in an educative manner.

**Comment by amankumarrr** [comment]
> A cool question would you ask this data? Be creative!

The relationship between the developer's first git commit and his arrival time

> How many systems would you be interacting with?
  - GitHub/Azure DevOps – issues, commits, 
  - Office automation log


> Would you need a Data Lake or Data Warehouse?

Data Warehouse

**Comment by JackDevAU** [comment]
> 1. A cool question would you ask this data? Be creative!

Find out a way to estimate a projects cost and time. Possibly even add a hot swappable dev tool to figure out the best devs for the project based on the projects needs, cost and time. 

> 2. How many systems would you be interacting with?

Accounting – Business expenses, client payment receipts, payment dates, etc.
CRM – Projects, skills, technologies, sales, leads, etc.

> 3.  Would you need a Data Lake or Data Warehouse?

Both!

**Comment by christoment** [comment]
> 1. A cool question would you ask this data? Be creative!

Developer effort vs traffic increase in website :)

> 2. How many systems would you be interacting with?

3/4

> 3. Would you need a Data Lake or Data Warehouse?

Data warehouse is essential, but will add Data Lake when Northwind is looking to expand

**Comment by jimmy1992abc** [comment]
> A cool question would you ask this data? Be creative!
Can data lakes, warehouses, bases, pots or bowls be able to eliminate duplicated data? 

> How many systems would you be interacting with?
4/6. No interaction with the system logs and website traffic yet.

> Would you need a Data Lake or Data Warehouse?
Definitely any of them, as long as they are able to use AI to structure the data.

**Comment by KahaMason** [comment]
> A cool question would you ask this data? Be creative!

What is the level of public interaction of a feature in relation to time & effort or development? What's the cost effectiveness in relation to the desired goal?

> How many systems would you be interacting with?

3 - GitHub/Azure DevOps, Timesheets, Website Analytics

> Would you need a Data Lake or Data Warehouse?

Data Warehouse

**Comment by jackreimers** [comment]
>A cool question would you ask this data? Be creative!  

The relationship between having people onsite vs remote and how that affects project delivery.

>How many systems would you be interacting with?  

Three.

>Would you need a Data Lake or Data Warehouse?

Data Warehouse.

**Comment by zacharykeeping** [comment]
> 1. A cool question would you ask this data? Be creative!

Is there a relationship between hours spent in office and increased productivity?

> 2. How many systems would you be interacting with?

2 - Office automation logs and GitHub/Azure DevOps

> 3. Would you need a Data Lake or Data Warehouse?

Data warehouse

**Comment by jernejk** [comment]
1.  A cool question would you ask this data? Be creative!
How many days do you have? Also, I'm more of a "What can I do with this thing" than "what can I ask it".

Ok, simple quick scenarios, we can fully automate daily scrum, sprint review, sprint planning, timesheets, and lots more. Have people review the generated stuff, add valuable insights outside of already existing knowledge and you have emails/documents that are painless to create, follow all of the rules to add the fluff that is already known from the system and valuable input from people that review these docs.

How? Ask questions like to Semantic Kernal:
- "What documents/entities/commits did the user create or changed yesterday?"
- "What tasks, bugs or PBIs were marked as done yesterday for this user? Are there any emails where the user responded with 'Done' to a task"?
- "What are the most urgent tasks/emails user X should tackle today?"

Still needs something at the end to clean up things that don't make sense but this can massively speed up the process and make it versatile to non-devs as well.

2. How many systems would you be interacting with?
Most of them. If you use Semantic Kernal with the data behind it, you can basically solve any problem you want with natural language. (I wouldn't necessarily recommend solving *everything* with this as that would be very expensive once you stabilize LLM responses and use a sufficiently powerful tier for this job. But you could solve annoying repetitive tasks requiring multiple steps in several different systems with a single prompt... Just saying)

4. Would you need a Data Lake or Data Warehouse?
Without AI, Data Warehouse sounds better. But is you use AI, Data Lake might actually give you more power.

**Comment by Dhruv-0987** [comment]
1. A cool question would you ask this data? Be creative!
How can we rate a skill of a specific employee based on how much they have worked with it in their current and past projects.
2. How many systems would you be interacting with?
3
GitHub/Azure DevOps – issues, commits, etc.
Billing - Timesheets, Client invoices, etc
CRM – Projects, skills, technologies, sales, leads, etc.
3. Would you need a Data Lake or Data Warehouse?
I would start out with a Data warehouse because of there only being 3 types of data.

**Comment by Ben0189** [comment]
1. A cool question would you ask this data? Be creative!
 What are the skills and technologies that contributed the most to the company's sales growth and success in the current year

2. How many systems would you be interacting with?
 3 or 4

3. Would you need a Data Lake or Data Warehouse?
  Data Warehouse

**Comment by tomek-i** [comment]
>A cool question would you ask this data? Be creative!
What is the distribution or amount between technical debt (bugs? refactors etc.) vs. value (features and usage of said feature).
And then drill down into finer details to associate a cost value using time spent etc.

>How many systems would you be interacting with?
the more the merrier

>Would you need a Data Lake or Data Warehouse?
Depends. Do we have big data or do we expect to grow into the scenario where this will be a must due to the amount of data we generate, or will a warehouse suffice for now and be more cost effective until we outgrow it ?

**Comment by piers-sinclair** [comment]
> 1. A cool question would you ask this data? Be creative!

I would like to compare client projects to see project technologies vs $ billed vs number of projects 

For example, 

| Technology | Billed  | Number of Projects | Billed per Project |
|------------|---------|-------------------|------------------|
| React      | $120k   | 8                 | $15,000          |
| Angular    | $56k    | 7                 | $8,000           |
| SQL Server | $150k   | 15                | $10,000          |
| Blazor     | $27.5k  | 5                 | $5,500           |
| Power Apps | $18k    | 3                 | $6,000           |


Then I would also like to **analyse the trend of the $ figure over time**

For example, we could see that React is 100k in 2021, 110k in 2022 and 120k in 2023 indicating an upward trend.

Then we could analyze which projects are delivering us the most value and determine which technology areas to focus on. In particular we could zone in on technologies that deliver high value but which have less projects in so we can try to delve more into that area.

> 2. How many systems would you be interacting with?

Timesheeting software
CRM

> 3. Would you need a Data Lake or Data Warehouse?

Maybe - It depends on the volume and scale of our operations as well as how extensive our reporting system is or is planned to be.

**Comment by jomelsontanco** [comment]
### A cool question would you ask this data? Be creative!

Feels like Big Brother but here are some:

**Bathroom Breaks per day per Git Commit** = (Number Employee Entrance /Exit Times per Day )/ Total Employee Git Commits
**$ Value per Employee Git Commit** = Total $ Value of Employee Client Invoices /  Total Employee Git Commits
**SugarLearning Items Completion Correlation with Employee Chargeability** = SugarLearning Items Completion Per Month AND Client Invoices Chargeabillity in the Next Three Months analysed over time

### How many systems would you be interacting with?

All of them.

### Would you need a Data Lake or Data Warehouse?

Both because:
in order to answer predefined questions and produce recurring reports, we need Data Warehouses.
in order to stimulate inspiration and boost innovation, we need Data Lakes.

**Comment by steven0x51** [comment]
1. A cool question would you ask this data? Be creative!
Does the working location have an impact on productivity?

2. How many systems would you be interacting with?
3

3. Would you need a Data Lake or Data Warehouse?
Data Warehouse

**Comment by bahjatalaadelssw** [comment]
I'd like to see a AI scanning plugin for data management that can scan images and videos using the query search words.

**Comment by PennyWalker** [comment]
I'd love for us to have a data lake/warehouse! We could get some amazing analytics out of it and really revolutionize our Power BI reports. Reading some of the ideas above is so inspiring! There are so many things we could do with it.

I've really loved learning about Data - my biggest Brisbane-based client is a massive data project and I've really enjoyed learning more about it! I feel like I can speak a little more about it now. :)

**Comment by camillars** [comment]
1. A cool question: What is the correlation between employee arrival/leave times and the productivity of different projects in the CRM system?

2. You would be interacting with six systems: GitHub/Azure DevOps, Billing, CRM, Accounting, Office automation logs, and Websites.

3. In this scenario, you would likely need both a Data Lake and a Data Warehouse. A Data Lake would be useful for storing raw and unprocessed data from all the systems, allowing for flexibility and scalability in data processing. It would be suitable for storing large volumes of data, such as the logs from office automation or website traffic data. On the other hand, a Data Warehouse would be necessary for organizing and structuring the data for analytical purposes, providing a centralized and structured repository for reporting, analysis, and data integration. It would be suitable for storing processed and aggregated data from multiple systems, such as sales data from the CRM system or financial data from the accounting system.
🤖 (checked by @matt-goldman 🧠)

**Comment by jontygardner** [comment]
> 1. A cool question would you ask this data? Be creative!
What is the level of productivity of employees on WFH days vs in-office? 

> 2. How many systems would you be interacting with?
4

>3. Would you need a Data Lake or Data Warehouse?
Data Warehouse (data lake when we have more employees e.g. 3000)

**Comment by AshAnil** [comment]
> 1. A cool question would you ask this data? Be creative!
Compare the productivity of employees using chatGPT and using Google search

> 2. How many systems would you be interacting with?
3

> 3. Would you need a Data Lake or Data Warehouse?
Data Warehouse

**Comment by stephanfakossw** [comment]
A cool question would you ask this data? Be creative!
Why is Prepaid in TimePro different to Xero
How many systems would you be interacting with?
5
Would you need a Data Lake or Data Warehouse?
Data warehouse

**Comment by babakamyljanovssw** [comment]
1. A cool question would you ask this data? Be Creative!
    How can you keep improving the analysis strategy?
2. How many systems would you be interacting with?
    3
3. Would you need a Data Lake or Data Warehouse?
    Data Warehouse

**Comment by JackDuan123** [comment]
my thoughts:
1. I will enjoy building data pipeline
2. I will love to see insights of data
3. most of the real insights of data is not exciting but somehow disappointing

My question:
No question
How many systems would you be interacting with?
3
Would you need a Data Lake or Data Warehouse?
Just data lake, I like dancing with binaries

**Comment by MattParkerDev** [comment]
1. A cool question would you ask this data? Be creative!
Who spends the most time speaking with clients?
2. How many systems would you be interacting with?
I dont know? One if you consider just the data warehouse?
3. Would you need a Data Lake or Data Warehouse?
Warehouse

**Comment by jerryluossw** [comment]
- A cool question would you ask this data? Be creative!
Whose one skill creates the highest profit?
- How many systems would you be interacting with?
4
- Would you need a Data Lake or Data Warehouse?
Data Warehouse

**Comment by jimmidier** [comment]
> A cool question would you ask this data? Be creative!  

What's the trends of consulting service requirements over the past year based on website traffic, client payments and crm opportunities, grouped by customer industries, and forecast till the end of the year?
> How many systems would you be interacting with?  

3
> Would you need a Data Lake or Data Warehouse?

Data Warehouse

**Comment by sylhuang** [comment]
> A cool question would you ask this data? Be creative!

Looking at the website traffic and decide which tech stacks are most visited/needed by our potential clients. Combining with our current client projects and invoice, come up with a skill set combination that's the most profitable.

> How many systems would you be interacting with?

3 or 4 depending on if Accounting is needed.

> Would you need a Data Lake or Data Warehouse?

Data Warehouse

**Comment by johnxu16** [comment]
> A cool question would you ask this data? Be creative!
What common characteristics do our long-term and short-term customers have? 
 
    > How many systems would you be interacting with?
3  

    > Would you need a Data Lake or Data Warehouse?
Data Warehouse

**Comment by Aibono1225** [comment]
> A cool question would you ask this data? Be creative!

What is the correlation between website traffic and sales revenue?

> How many systems would you be interacting with?

4

> Would you need a Data Lake or Data Warehouse?

Data Warehouse

**Comment by tino-liu** [comment]
>1. A cool question would you ask this data? Be creative!

Does more code commits or more lines of code generate more value and revenue? 
Would be cool to prove that using lines-of-code isn't exactly a good metric.

>2. How many systems would you be interacting with?

3 systems

>3. Would you need a Data Lake or Data Warehouse?

Data Warehouse

**Comment by JaneYangssw** [comment]
> 1.A cool question would you ask this data? Be creative!

Can we use traffic and analytics data from the websites to identify opportunities for new business or marketing strategies?

> 2.How many systems would you be interacting with?

Three

> 3.Would you need a Data Lake or Data Warehouse?

Data Warehouse

**Comment by AttackOnMorty** [comment]
> 1. A cool question would you ask this data? Be creative!

Find topics for knowledge sharing (Hold session or CTF)
- The skills people want to learn the most
- The most popular technologies among all current client projects

> How many systems would you be interacting with?

3 (Billing, CRM, Accounting)

> 3. Would you need a Data Lake or Data Warehouse?

Data Warehouse

**Comment by SofieHong** [comment]
1. A cool question would you ask this data? Be creative!
If you could record all human behaviour and convert it into visualized data, what interesting trends and patterns do you think you would see?

2. How many systems would you be interacting with?
3

3. Would you need a Data Lake or Data Warehouse?
Data Warehouse

**Comment by hveraus** [comment]
1. A cool question would you ask this data? Be creative!
Does people just back from holiday have higher or lower productivity than before taking the holiday. e.g. more PRs done?

2. How many systems would you be interacting with?
Maybe 3

3. Would you need a Data Lake or Data Warehouse?
N/A

**Comment by KristenHu** [comment]
```
1. A cool question would you ask this data? Be creative!
```
How many complaints would we receive from our clients every month🧐
```
3. How many systems would you be interacting with?
```
4 or 5 I think.
```
4. Would you need a Data Lake or Data Warehouse?
```
Data Warehouse

**Comment by AnkyChe** [comment]
> 1. A cool question would you ask this data? Be creative!
What skill is easy to learn but very profitable?

> 2. How many systems would you be interacting with?
3 I think

> 3. Would you need a Data Lake or Data Warehouse?
Data Warehouse

**Comment by Freego1783** [comment]
Data engineering looks really promising, I'm keen to see what insights we can get out of SSW's data !

**Comment by PothieuG** [comment]
1. A cool question would you ask this data? Be creative!
Can we use traffic from the application to create new marketting strategies ?

2. How many systems would you be interacting with?
Three

3. Would you need a Data Lake or Data Warehouse?
Data Warehouse

**Comment by BrookJeynes** [comment]
> A cool question would you ask this data? Be creative!

Are people who work less hours more productive in the hours they work vs people who work more hours

> How many systems would you be interacting with?

Maybe 3

> Would you need a Data Lake or Data Warehouse?

Data Warehouse

**Comment by sethdaily** [comment]
> A cool question would you ask this data? Be creative!
Does the frequency of GitHub/DevOps commits on a project correlate with client satisfaction?
  > How many systems would you be interacting with?
2 - GitHub/Azure for commits, and CRM for projects and client satisfaction data
  > Would you need a Data Lake or Data Warehouse?
I think you would need a data lake, because this would be a unique set of queries that the company probably does not have a data warehouse for (you could probably build a data warehouse that does this too).

**Comment by leahy268** [comment]
A cool question would you ask this data? Be creative!
I would be interested in productivity versus office/remote work probably correlated with customer satisfaction and developer satisfaction.

How many systems would you be interacting with?
Three

Would you need a Data Lake or Data Warehouse?
Not sure

**Comment by UlyssesMaclaren** [comment]
> A cool question would you ask this data? Be creative!
What referral sources brought us our main clients?
Profit and loss per month /quarter?
Which developers get the most work done?

 > How many systems would you be interacting with?
All of them (6)

 > Would you need a Data Lake or Data Warehouse?
Data Warehouse as all the data looks structured

**Reply by tomek-i** [reply]
> Which developers get the most work done?

Hmm. What do you consider as more work? The amount of PBI's completed? The amount of lines changed? Something different?


* Amount of PBI's can be misleading, I could have 10 PBI's with super small fixes like spelling vs, 1 PBI that introduced a new feature with 1000 of lines. 
* Amount of lines can also be misleading because I could have 10.000 lines changed where I did a simple find & replace where I renamed a variable. vs troubleshooting for 2 days a bug until finding the root cause to be a logical error. Fixing this could be changing one line.

Perhaps the question would be, which of the work had the most value to the business? That would make sense, however there are also many scenarios that can occur that skew it. For example consider the scenario that a developer picks unassigned PBI's from the print / backlog and has a streak of picking low value PBIs? 


Finding good and consistent correlation between all of these variables is the difficult part :-)

**Comment by leehawkinsssw** [comment]
1. How does the number and severity of bugs vary based on developer seniority and time of day?
2. 2-3
3. Data warehouse would suffice (data to support answering this question feels structured in nature)

**Comment by Marxoz** [comment]
> A cool question would you ask this data? Be creative!
Based on the enquiries we receive through the website, what are the main technologies requested. 
And how does that compare to the existing projects that we are working on. Break it up into number of projects as well as amount of $.

   > How many systems would you be interacting with?
3 

   > Would you need a Data Lake or Data Warehouse?
Although a Data warehouse should be efficient, a Data lake will make more sense. As you answer the questions above, you will start building patterns and ask more questions, which will require more data. This data might only be know once you have answered the question above, so a Data Lake might be more useful in the long run

**Comment by chrisschultzssw** [comment]
> A cool question would you ask this data? Be creative!

Which lead sources end up being the most profitable?
Similarly - which skills/tech are the most profitable?

> How many systems would you be interacting with?

3

> Would you need a Data Lake or Data Warehouse?

Data warehouse

**Comment by Hona** [comment]
> A cool question would you ask this data? Be creative!

Does the device you choose to work with (Windows/Mac etc), impact any metrics of statistical significance?

> How many systems would you be interacting with?

Possibly all of them

> Would you need a Data Lake or Data Warehouse?

Data Warehouse.

**Comment by Levijj22** [comment]
Incredible video! What a great explanation.

**Comment by KostaMadorsky** [comment]
1. Cost per line of code 🙃
2. Gotta Pull ’Em All
3. DW
